### DISCUSSION TITLE
OpenTelemetry as a file usage logger

### DISCUSSION THREAD

**Discussion by ovi-s** [open-telemetry/opentelemetry-collector#3442] [Q&A] [answered]
Hello,
I have a software that uses certain types of files. I want to have a record when a certain file was used and by who. For this, I want to make a setup something like this:
1) Raise a span in my program (which is written in C++) with the file name and the username.
2) Send it to a remote collector via OTLP
3) The collector receives the data and send it as a http request to some server that handles this request and put the data into a database

Is OpenTelemetry suitable for this scenario? If yes, I will put the rest of my questions here :)

**Comment by bogdandrutu** [comment, marked as answer]
So far yes :) since you create spans and transfer via the collector.
